src/confluent_kafka/src/AdminTypes.c

```diff
@@ -125,7 +139,8 @@ static PyMemberDef NewTopic_members[] = {
         { "topic", T_STRING, offsetof(NewTopic, topic), READONLY,
           ":py:attribute:topic - Topic name (string)" },
         { "num_partitions", T_INT, offsetof(NewTopic, num_partitions), 0,
-          ":py:attribute: Number of partitions (int)" },
+          ":py:attribute: [OPTIONAL] Number of partitions (int).\n"
+          "Must be set to -1 or not be provided if a replica_assignment is specified" },
```

Comment:
Check this - https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/NewTopic.java